minoc/src/intTest/java/org/opencadc/minoc/SetStateTest.java

```diff
@@ -0,0 +1,397 @@
+
+            // update
+            // TODO: add update to artifactURI when functionality available
+            String newEncoding = "test-encoding-2";
+            String newType = "application/x-text-message";
+            Map<String,Object> postParams = new HashMap<String,Object>(2);
+            postParams.put("contentEncoding", newEncoding);
+            postParams.put("contentType", newType);
+            HttpPost post = new HttpPost(artifactURL, postParams, false);
+            post.setDigest(computeChecksumURI(data));
+            Subject.doAs(userSubject, new RunnableAction(post));
+            Assert.assertNotNull("post to minoc in offline state should have failed", post.getThrowable());
+            exMsg = post.getThrowable().getMessage();
+            Assert.assertTrue("incorrect exception message from post", exMsg.contains("System is offline"));
+
+            // head
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            HttpGet head = new HttpGet(artifactURL, bos);
+            Subject.doAs(userSubject, new RunnableAction(head));
+            Assert.assertNotNull("head from minoc in offline state should have failed", head.getThrowable());
+            exMsg = head.getThrowable().getMessage();
+            Assert.assertTrue("incorrect exception message from head", exMsg.contains("System is offline"));
+
+            // delete
+            HttpDelete delete = new HttpDelete(artifactURL, false);
+            Subject.doAs(userSubject, new RunnableAction(delete));
+            Assert.assertNotNull("delete from minoc in offline state should have failed", delete.getThrowable());
+            exMsg = delete.getThrowable().getMessage();
+            Assert.assertTrue("incorrect exception message from delete", exMsg.contains("System is offline"));
+        } catch (Exception t) {
+            log.error("unexpected throwable", t);
+            Assert.fail("unexpected throwable: " + t);
+        } finally {
+	    // restore minoc to working state
+            String state = RestAction.STATE_READ_WRITE;
+	    setState(state);
+	}
+    }
+            
+    @Test
+    public void testSetReadOnly() throws Exception {
+        try {
+            // set minoc state to 'ReadWrite'
+            String state = RestAction.STATE_READ_WRITE;
+	    setState(state);
+
+            // test get action
+            URI artifactURI = URI.create("cadc:TEST/file.txt");
+            URL artifactURL = new URL(filesURL + "/" + artifactURI.toString());
+
+            String content = "abcdefghijklmnopqrstuvwxyz";
+            String encoding = "test-encoding";
+            String type = "text/plain";
+            byte[] data = content.getBytes();
+
+            // put a file so that we can test get: no length or checksum
+            InputStream in = new ByteArrayInputStream(data);
+            HttpUpload put = new HttpUpload(in, artifactURL);
```

Comment:
Since this put is intended to be successful, the one header that should always be set is content-length (it may effect server side storage strategy in future). All the others are just noise as noted above.

Reply:
Added content-length to put.